### DISCUSSION TITLE
Where to write logic if iModel Viewer re-render ?

### DISCUSSION THREAD

**Discussion by ajay-toobler** [iTwin/itwinjs-core#4316] [Q&A]
In my observation through the samples I have seen.

- We could write mounting logic inside `onImodelConnect`. Which won't re-excute on iModel update.
- If we have to append some UI with logic . we could use of `uiProviders`. which will re-render on iModel update.

What if I want to write some code which don't have any jsx but something like a **custom hook** which needs to re-render on iModel update ?

```
<Viewer
       {...otherProps}
        uiProviders={[
           ...uiProviders,
        ]}
        onIModelConnected={onIModelConnected}
      />
```

**Comment by ayushibasaiwal** [comment]
@pmconne @aruniverse Can you please look into this?

**Comment by pmconne** [comment]
@ajay-toobler please be specific about what you need / what you are trying to accomplish / what you mean by "iModel update".
